File: Backend/SistemaVendas.Api/Models/Venda.cs
namespace SistemaVendas.Api.Models;

public class Venda
{
    public int Id { get; set; }

    public int ClienteId { get; set; }
    public Cliente? Cliente { get; set; }

    public DateTime DataVenda { get; set; } = DateTime.UtcNow;

    public string Status { get; set; } = "Pendente";

    public decimal ValorTotal { get; set; }

    public List<ItemVenda> Itens { get; set; } = new();
}

File: Frontend/marketplace-front/types/venda.ts
export type CriarVendaRequest = {
  clienteId: number
}

export type AdicionarItemVendaRequest = {
  produtoId: number
  quantidade: number
}

export type VendaResponse = {
  id: number
  clienteId: number
  dataVenda?: string
  status?: string
  valorTotal?: number
}